### PR TITLE
Add a workflow for basic authentication

### PR DESCRIPTION
**Pull request by vermiculus**
Example usage:

```lisp
(defun ghub-new-authentication (scopes note)
  "Authenticate with SCOPES using NOTE."
  (let ((ghub-authenticate 'basic))
    (cdr (assq 'token (ghub-post "/authorizations" nil
                                 `((scopes . ,scopes)
                                   (note . ,note)))))))
```

Addresses #4.

### CODE REVIEW

File: ghub.el [hunk 1]
```diff
@@ -132,8 +139,8 @@
          (d (and data   (json-encode-list data)))
          (url-request-extra-headers
           `(("Content-Type"  . "application/json")
-            ,@(when-let ((token (ghub--token)))
-                `(("Authorization" . ,(concat "token " token))))))
+            ,@(and ghub-authenticate
```
Comment (vermiculus) [inline ghub.el]: The check for `ghub-authenticate` moved up here

File: ghub.el [hunk 2]
```diff
@@ -182,17 +189,28 @@
...
+                  (ghub--token)))
+         url response)
+    (if token
+        (concat "token " token)
+      (setq url (url-generic-parse-url ghub-base-url))
+      (setf (url-user url) ghub-username)
+      (url-basic-auth url t))))
+
 (defun ghub--token ()
-  (and ghub-authenticate
```
Comment (vermiculus) [inline ghub.el]: The check for `ghub-authenticate` moved up to `ghub--request`.

**Comment by tarsius** [comment]
Please move the typo fixes to a separate commit.

**Comment by vermiculus** [comment]
Done, but I'm not sure what GitHub is thinking. Might be best to merge it with Magit.

Here's what the log says for me:
```
0639508 * basic-auth-flow vermiculus/basic-auth-flow Add a workflow for basic authentication
092f6c6 * Fix typos
da60fa2 * v1.2.0 origin/master origin/HEAD master Mention apiwrap.el in documentation
```

**Comment by vermiculus** [comment]
Fixed.

**Comment by tarsius** [comment]
> I'm not sure what GitHub is thinking

Github's log is *very* basic. Among other limitations, it just sorts by author data, I believe.

**Comment by tarsius** [comment]
Closing in favor of #11.

**Pull request closed without merging**